### PR TITLE
Add test for project model and view

### PR DESCRIPTION
**Pull request by alicoding**

### CODE REVIEW

**Comment by alicoding** [comment]
@gideonthomas R?

File: app/scienceapi/projects/tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        lambda o: '{a}-institution'.format(a=o.name.lower())
+    )
+    description = factory.LazyAttribute(
+        lambda o: 'Long description for {a}'.format(a=o.name.lower())
+    )
+    short_description = factory.LazyAttribute(
+        lambda o: 'Short description for {a}'.format(a=o.name.lower())
+    )
+    status = True
+    date_created = datetime.date.today()
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_models.py]: hmm you shouldn't need to add these date fields

File: app/scienceapi/projects/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    date_updated = datetime.date.today()
+    license = "MIT"
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_models.py]: single quotes

File: app/scienceapi/projects/tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    class Meta:
+        model = Project
+
+
+class UserFactory(factory.Factory):
+    username = 'john'
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_models.py]: why are these hardcoded?

File: app/scienceapi/projects/tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    email = factory.LazyAttribute(lambda o: '%s@example.com' % o.username)
+    location = factory.LazyAttribute(lambda o: 'Name!')
+    biography = factory.LazyAttribute(lambda o: 'Name!')
+    github_id = factory.Sequence(lambda n: str(n))
+    github_username = factory.LazyAttribute(lambda o: 'Name!')
+    twitter_handle = factory.LazyAttribute(lambda o: 'Name!')
+    avatar_url = factory.LazyAttribute(lambda o: 'Name!')
+    blog = factory.LazyAttribute(lambda o: 'Name!')
+    company = factory.LazyAttribute(lambda o: 'Name!')
+    date_created = factory.LazyAttribute(lambda o: 'Name!')
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_models.py]: You shouldn't need to create these

File: app/scienceapi/projects/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    role = factory.LazyAttribute(lambda o: 'Name!')
+
+    class Meta:
+        model = User
+
+
+class UserProjectFactory(factory.Factory):
+    user = factory.SubFactory(UserFactory)
+    project = factory.SubFactory(ProjectFactory)
+    role = 'something'
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_models.py]: `Contributor`

File: app/scienceapi/projects/tests/test_views.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import json
+
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from scienceapi.projects.serializers import ProjectSerializer
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_views.py]: let's have a blank line before this

File: app/scienceapi/projects/tests/test_views.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+class TestProjectListView(TestCase):
+
+    def test_list_projects_returns_project_data(self):
+        user = UserFactory()
+        user.save()
+        project = ProjectFactory()
+        project.save()
+        link = ResourceLinkFactory.build(
+            project=project,
```
Comment (gideonthomas) [inline app/scienceapi/projects/tests/test_views.py]: should be able to put this in the previous line

**Pull request merged**